joplin/base/models/janis_page.py

```diff
@@ -168,8 +167,12 @@ def janis_preview_url_end(self, revision=None):
         else:
             revision = self.get_latest_revision()
             url_page_type = self.janis_url_page_type
-        global_id = graphene.Node.to_global_id('PageRevisionNode', revision.id)
-        url_end = f"preview/{url_page_type}/{global_id}"
+        try:
+            global_id = graphene.Node.to_global_id('PageRevisionNode', revision.id)
+            url_end = f"preview/{url_page_type}/{global_id}"
+        except AttributeError:
+            url_end = f"preview/{url_page_type}/"
```

Comment:
Sorry I forgot, when does this happen again? If a page doesn't have a revision yet? Do locations not have revisions?

If this exception has been added just for the sake of MVP locations, let's make a "TODO" note to clean up previews on locations and then get rid of this logic. Because a preview of a page that has no revision should never exist.

Reply:
IIRC it had to do with loading fixtures or generating test pages, those pages exist but I think they didn't have revisions associated with them yet. So it could happen with any page that is created but not with an associated revision. 

Example test page:
https://joplin-pr-3296-location-mvp-mv.herokuapp.com/admin/pages/267/revisions/

A location page has revisions like any other Wagtail Page type. 

Since we're writing our own url logic here, I think it's our responsibility to handle edge cases. In this case I wouldn't expect this to ever be a problem in staging or prod unless we progamatically enter content at some point(which could very well happen), but the try statement here has no performance drawbacks but helps avoid unexpected problems that might break the admin. 

From my perspective, good candidates for refactoring in this case would be how we handle previews (using https://github.com/torchbox/wagtail-headless-preview would be nice) and re-tooling to leverage wagtails existing functions for building urls which are more robust (or least informing our design based on them: 
https://github.com/wagtail/wagtail/blob/8238a338888258554ed6eae0e25eaf903a0a5831/wagtail/core/models.py#L748 )

Reply:
But this doesn't handle the edge case. It's just going to allow previews to break for test fixtures.

If fixtures don't have revision data, that should be accounted for in logic farther up the line rather than returning a preview_url that won't work.

But for the sake of getting the locations MVP merged in, let's worry about that later.

Reply:
Yes, to be clear, it doesn't make previews work(which I didn't really have in scope for MVP as far as I'm aware), but without this try statement the entire build was breaking.